### PR TITLE
Allow legacy user id migration and bearer token auth for migration endpoint

### PR DESCRIPTION
**Pull request by toddkazakov**
- Add the ability to mirgate a legacy user id to keycloak - if the user id is not present in the response a keycloak uuid will be automatically generated 
- Add the ability to secure requests to the migration endpoint using a bearer token
- Bump Keycloak to 11.0.0

### CODE REVIEW

**Comment by daniel-frak** [comment]
Hmm, looks like I bungled something up when configuring SonarCloud for this project. I'll try to fix this today!

**Comment by daniel-frak** [comment]
Could you try pulling the latest changes? It should resolve the CI issues.

File: README.md [hunk 1]
```diff
@@ -117,6 +127,15 @@ Additional configuration options are available for fine-tuning the migration.
 
 ![Additional configuration](readme-images/config.png)
 
+### API Token
+
+The migration endpoint can be secured with an API token. The configured value will be sent as a bearer token in the authorization header.
+
+If the configured token value is set to `SECRET_API_TOKEN` when making the request to the migration endpoints, the rest client will send the following authroization header:
```
Comment (daniel-frak) [inline README.md:134]: `authroization`→`authorization`

File: src/main/java/com/danielfrak/code/keycloak/providers/rest/remote/UserModelFactory.java [hunk 2]
```diff
@@ -51,7 +52,22 @@ public UserModelFactory(KeycloakSession session, ComponentModel model) {
...
+            userModel = session.userLocalStorage().addUser(
+                realm,
+                legacyUser.getId(),
+                legacyUser.getUsername(),
+                true,
+                false
+            );
+        }
+
+        UserRepresentation rep = new UserRepresentation();
```
Comment (daniel-frak) [inline src/main/java/com/danielfrak/code/keycloak/providers/rest/remote/UserModelFactory.java:68]: This doesn't seem to be doing anything useful. Am I wrong?

**Comment by daniel-frak** [comment]
Yes, it works! :)

Hmm, do you have time to add some tests for `BearerTokenRequestFilter`? If not, that's fine, as I kind of added the SonarCloud check half-way through the PR, just let me know.

The rest of the classes you can ignore, as their lack of coverage is my fault - I'll add more tests to them later, after I merge in your code.

Thanks again for the contribution!

**Comment by toddkazakov** [comment]
@daniel-frak added tests for `BearerTokenRequestFilter`

**Comment by daniel-frak** [comment]
Thanks a lot!

**Pull request merged**